lib/practice_exercises.rb

```diff
@@ -1,13 +1,50 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n)
+# Space Complexity: o(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  string_array = string.split("") 
+  alpha_only = []
+  alpha_copy = []
+  pattern = /^[A-Za-z]+$/
+  string_array.each do |char|
+    if char.match?(pattern)
+      alpha_only << char.downcase 
+      alpha_copy << char.downcase
+    end 
+  end 
+  length = alpha_only.length
+  (length/2).times do |index|
+    temp = alpha_only[index]
+    alpha_only[index] = alpha_only[length - 1 - index]
+    alpha_only[length - 1 - index] = temp
+  end 
+  return alpha_only.join() == alpha_copy.join()
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: o(n^2)
+# Space Complexity: o(1)
 def longest_prefix(strings)
```

Comment:
I would say the time complexity is O(n * m) where m is the length of the shortest word and n being the number of strings.  Otherwise 👍